e2e/cephfs.go

```diff
@@ -126,14 +129,53 @@ var _ = Describe("cephfs", func() {
 					Fail(err.Error())
 				}
 			}
+			if upgradeTesting {
+				// fetch current working directory to switch back
+				// when we are done upgrading.
+				cwd, err := os.Getwd()
+				if err != nil {
+					Fail(err.Error())
+				}
+				err = upgradeAndDeployCSI(upgradeVersion, "cephfs")
+				if err != nil {
+					Fail(err.Error())
+				}
+				createConfigMap(cephfsDirPath, f.ClientSet, f)
+				createCephfsSecret(f.ClientSet, f)
+				createCephfsStorageClass(f.ClientSet, f, true, "")
+				pvcPath := cephfsExamplePath + "pvc.yaml"
+				appPath := cephfsExamplePath + "pod.yaml"
+				_, app := createPVCAndAppBinding(pvcPath, appPath, f, deployTimeout)
+				err = deletePod(app.Name, app.Namespace, f.ClientSet, deployTimeout)
+				if err != nil {
+					Fail(err.Error())
+				}
+				deleteCephfsPlugin()
+
+				// support of PSP is available in current master, so we don't need to
+				// skip it now.
+				skipPSP = false
+				err = os.Chdir(cwd)
+				if err != nil {
+					Fail(err.Error())
+				}
+				deployCephfsPlugin()
+				// validate if the app gets bound to a pvc created by
+				// an earlier release.
+				err = createApp(f.ClientSet, app, deployTimeout)
+				if err != nil {
+					Fail(err.Error())
+				}
+				Skip("Skipping CephFS E2E")
+			}
 			deployCephfsPlugin()
 		}
 		createConfigMap(cephfsDirPath, f.ClientSet, f)
 		createCephfsSecret(f.ClientSet, f)
 	})
 
 	AfterEach(func() {
-		if !testCephFS {
+		if !testCephFS || upgradeTesting {
```

Comment:
why check for upgradeTesting here?

Reply:
This was added to skip cephFS e2e, in upgrade testing job. (to avoid redundant checks)